rewriter/Initializer.cc

```diff
@@ -29,6 +29,30 @@ bool isCopyableType(const ast::ExpressionPtr &typeExpr) {
     return true;
 }
 
+// Not checking for being T.proc, it's expected to be checked
+void maybeRemoveBind(core::Context ctx, ast::Send *send) {
+    auto lastSend = send;
+    while (send != nullptr) {
+        if (send->fun == core::Names::bind()) {
+            // `bind` is the last send in the chain of sends.
+            // This is incorrect syntax which is covered by the resolver.
+            // Safe to ignore.
+            if (send == lastSend) {
+                return;
+            }
+            auto recvSend = ast::cast_tree<ast::Send>(send->recv);
+            if (recvSend != nullptr) {
+                lastSend->recv = move(send->recv);
+                return;
+            }
+            // We handle only one `.bind` call in the whole chain,
+            // no need to traverse the rest of the tree.
+            return;
```

Comment:
And this as well?

Reply:
Same as above with these two errors competing for the same line of code:
- Malformed `bind`: Multiple calls to `.bind`
- Using `bind` is not permitted here

Reply:
We have docs on how to add multiple errors for a single line: https://github.com/sorbet/sorbet/#test_corpus-tests

Reply:
+1 to drive home what nathan mentions:

```
T.proc.bind(A).bind(A).void
#      ^^^^ error: first error
#              ^^^^ error: second error
```

should let you make these assertions